src/EgressClient.ts

```diff
@@ -293,10 +293,10 @@ export class EgressClient extends ServiceBase {
     let preset: EncodingOptionsPreset | undefined;
     let advanced: EncodingOptions | undefined;
 
-    if ((<EncodedOutputs>output).fileOutput !== undefined) {
-      file = (<EncodedOutputs>output).fileOutput;
-      stream = (<EncodedOutputs>output).streamOutput;
-      segments = (<EncodedOutputs>output).segmentOutput;
+    if ((<EncodedOutputs>output).file !== undefined) {
```

Comment:
actually, noticed an issue here. this wouldn't stick if file isn't set but only stream is. you should break those out into separate if elses..